frontend/js/src/playlists/SearchDropDown.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from "react";
+import { faUser } from "@fortawesome/free-solid-svg-icons";
+import GlobalAppContext from "../utils/GlobalAppContext";
+import ListenControl from "../listens/ListenControl";
+
+export type SearchDropDownProps = {
+  addCollaborator: (event: string) => void;
+  userSearchResults: Array<any>;
+};
+
+export type SearchDropDownState = {};
+
+export default class SearchDropDown extends React.Component<
```

Comment:
I think this component should also be made more generic so that it can be used in other places where we will need to search for a user, and it should handle all the logic for searching for  and selecting a user.

I suggest to name the component `UserSearch` and have it handle both the API search *and* rendering the results (which you already do here).
It should have a callback prop `onUserSelect` instead of `addCollaborator` so that it is more generic.

I think you will make your life a lot easier by using the react-select library which we already use in other parts of the codebase.
It renders an input and the dropdown and everything else you need.
Have a look here for example: https://github.com/metabrainz/listenbrainz-server/blob/4b3fa44f7555c5ba8dc28e7d7cbf5b2a6c5b51c8/frontend/js/src/playlists/Playlist.tsx#L807-L821